### ISSUE TITLE
Secret handshake results in error UnhandledPromiseRejectionWarning

### ISSUE THREAD

**Issue by dandrewstewart** [michaelpalumbo/wspd#10]
Choose the server type: Secret Handshake is generating error after completing all initialisation steps.

`(node:5419) UnhandledPromiseRejectionWarning: ReferenceError: dap is not defined
    at /usr/local/lib/node_modules/allhands/bin/allhands.js:228:51
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:5419) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:5419) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.`

**Comment by michaelpalumbo** [comment]
Thanks @dandrewstewart 

I'll take a look

**Comment by dandrewstewart** [comment]
> Choose the server type: Public works fine, however 
Could you elaborate? I'm not sure what you meant by this, thanks!

**Comment by michaelpalumbo** [comment]
@dandrewstewart I believe I have fixed the problem. Please reinstall allhands to update to this version ```npm install -g allhands```

**Comment by michaelpalumbo** [comment]
> Choose the server type: Public works fine, however 

Could you elaborate? I'm not sure what you meant by this, thanks!

Also I made an error and edited your previous comment above instead of quote-replying, and I cannot get your original comment back sorry about that.

**Comment by djiamnot** [comment]
I seem to have reached the same or related issue:
```
~/.local/share/nvm/v16.13.2/lib/node_modules/allhands/bin/allhands.js:250
      config.set(configFileName + '.thisNode.dap', dap)
                                                   ^

ReferenceError: dap is not defined
    at ~/.local/share/nvm/v16.13.2/lib/node_modules/allhands/bin/allhands.js:250:52
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```
Both seem to concern `dap` not defined.
And to echo @dandrewstewart, I get this error when I choose "secret handshake" as the server type. Choosing "Public server" works.

**Comment by michaelpalumbo** [comment]
@djiamnot sorry to take a week to get back to you, I was quite busy last week. I'll look into this today!

**Comment by djiamnot** [comment]
No worries. You've been very helpful and reactive, you've got some karma credits, AFAIC :zany_face: 
We've been moving slowly, anyway, so this particular issue wasn't blocking.

**Comment by michaelpalumbo** [comment]
Haha, awesome, thanks @djiamnot

**Comment by michaelpalumbo** [comment]
@djiamnot @dandrewstewart Error is fixed now, with version 0.0.74, so you'll want to reinstall via ```npm install -g allhands``` It was a typo, left over from the version prior to storing/recalling configs.

**Comment by michaelpalumbo** [comment]
closing this issue :)
